internal/exec/resolvable/resolvable.go

```diff
@@ -228,13 +228,17 @@ func (b *execBuilder) makeObjectExec(typeName string, fields schema.FieldList, p
 
 	Fields := make(map[string]*Field)
 	rt := unwrapPtr(resolverType)
+	fieldsCount := getFieldCount(rt, map[string]int{})
 	for _, f := range fields {
-		fieldIndex := -1
+		var fieldIndex []int
```

Comment:
Why does the field index is an array? I don't get it.

Reply:
It is an array because it is an index sequence for [`Type.FieldByIndex`](https://golang.org/pkg/reflect/#Value.FieldByIndex).

It allows for several layers of anonymously-nested fields.